front_end/panels/ai_chat/core/EnvironmentConfig.ts

```
@@ -0,0 +1,337 @@
+  private initializeFromRuntime(): void {
+    if (!this.runtimeConfig) {
+      return;
+    }
+
+    const providers: APIKeyProvider[] = ['openai', 'openrouter', 'groq', 'litellm'];
+    let savedCount = 0;
+
+    for (const provider of providers) {
+      const storageKey = STORAGE_KEYS[provider];
+      const existingKey = localStorage.getItem(storageKey);
+      
```

ESLint

[error] Trailing spaces not allowed.

(@stylistic/no-trailing-spaces)